lib/resque/failure/redis.rb

```diff
@@ -46,7 +50,7 @@ def self.each(offset = 0, limit = self.count, queue = :failed, class_name = nil,
           reversed = true
         end
         all_items.each_with_index do |item, i|
-          if !class_name || (item['payload'] && item['payload']['class'] == class_name)
+          if !class_name || (item['payload'] && item['payload']['class'] == class_name && (original_limit -= 1) >= 0)
```

Comment:
`original_limit` acts as a counter here, `original_limit - 1` will produce the same result on each loop.